OPAL/tac/src/main/scala/org/opalj/tac/fpcf/analyses/string/StringAnalysis.scala

```diff
@@ -0,0 +1,287 @@
+package fpcf
+package analyses
+package string
+
+import scala.collection.mutable.ListBuffer
+
+import org.opalj.br.DeclaredMethod
+import org.opalj.br.Method
+import org.opalj.br.analyses.SomeProject
+import org.opalj.br.fpcf.ContextProviderKey
+import org.opalj.br.fpcf.FPCFAnalysis
+import org.opalj.br.fpcf.analyses.ContextProvider
+import org.opalj.br.fpcf.properties.Context
+import org.opalj.br.fpcf.properties.cg.Callers
+import org.opalj.br.fpcf.properties.cg.NoCallers
+import org.opalj.br.fpcf.properties.string.StringConstancyProperty
+import org.opalj.br.fpcf.properties.string.StringTreeNode
+import org.opalj.fpcf.EOptionP
+import org.opalj.fpcf.EPK
+import org.opalj.fpcf.EPS
+import org.opalj.fpcf.EUBP
+import org.opalj.fpcf.InterimResult
+import org.opalj.fpcf.ProperPropertyComputationResult
+import org.opalj.fpcf.Result
+import org.opalj.fpcf.SomeEPS
+import org.opalj.fpcf.UBP
+import org.opalj.log.Error
+import org.opalj.log.Info
+import org.opalj.log.OPALLogger.logOnce
+import org.opalj.tac.fpcf.properties.TACAI
+import org.opalj.tac.fpcf.properties.string.MethodStringFlow
+
+trait StringAnalysis extends FPCFAnalysis {
+
+    private final val ConfigLogCategory = "analysis configuration - string analysis"
+
+    protected val maxDepth: Int = {
+        val maxDepth =
+            try {
+                project.config.getInt(StringAnalysis.MaxDepthConfigKey)
+            } catch {
+                case t: Throwable =>
+                    logOnce(Error(ConfigLogCategory, s"couldn't read: ${StringAnalysis.MaxDepthConfigKey}", t))
+                    30
+            }
+
+        logOnce(Info(ConfigLogCategory, "using maximum depth " + maxDepth))
+        maxDepth
+    }
+}
+
+object StringAnalysis {
+
+    final val MaxDepthConfigKey = "org.opalj.fpcf.analyses.string.StringAnalysis.maxDepth"
+}
+
+/**
+ * @author Maximilian Rüsch
+ */
+private[string] class ContextFreeStringAnalysis(override val project: SomeProject) extends StringAnalysis {
+
+    def analyze(vd: VariableDefinition): ProperPropertyComputationResult =
+        computeResults(ContextFreeStringAnalysisState(vd, ps(vd.m, MethodStringFlow.key)))
+
+    private def continuation(state: ContextFreeStringAnalysisState)(eps: SomeEPS): ProperPropertyComputationResult = {
+        eps match {
+            case _ if eps.pk == MethodStringFlow.key =>
+                state.stringFlowDependee = eps.asInstanceOf[EOptionP[Method, MethodStringFlow]]
+                computeResults(state)
+
+            case _ =>
+                throw new IllegalArgumentException(s"Unexpected eps in continuation: $eps")
+        }
+    }
+
+    private def computeResults(implicit state: ContextFreeStringAnalysisState): ProperPropertyComputationResult = {
+        val newProperty = StringConstancyProperty(state.stringFlowDependee match {
+            case UBP(methodStringFlow) =>
+                val tree = methodStringFlow(state.entity.pc, state.entity.pv)
+                if (tree.depth == maxDepth) {
```

Comment:
Are you sure depth can never "jump" more than one so that depth > maxDepth and the check here doesn't trigger?